internal/provider/data_source_project.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/preview/2019-12-10/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var projectDefaultTimeout = time.Minute * 1
+
+func dataSourceProject() *schema.Resource {
+	return &schema.Resource{
+		Description: "The project data source provides information about an existing HashiCorp project.",
+		ReadContext: dataSourceProjectRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &projectDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"project_id": {
+				Description:      "The ID of the HCP project.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
```

Comment:
⚪️ The project ID should be expected to be a UUID not a SlugID. We might however want to skip client side validation here to allow us to easier switch to a different identifier format in the future.